sbncode/FlashMatch/FlashPredict_module.cc

```diff
@@ -566,113 +558,127 @@ void FlashPredict::loadMetrics()
-    pe_spreads.push_back(0.001);
+    fRatioMeans.push_back(0);
+    fRatioSpreads.push_back(0.001);
   }
   else {
-    // std::vector<double> x, yL, yM, yH;
     for (int ib = 1; ib <= bins; ++ib) {
       double me = temphisto->GetBinContent(ib);
-      pe_means.push_back(me);
-      // x.push_back(temphisto->GetBinCenter(ib));
+      fRatioMeans.push_back(me);
       double tt = temphisto->GetBinError(ib);
       if (tt <= 0) {
-        std::cout << "zero value for bin spread in pe" << std::endl;
+        std::cout << "zero value for bin spread in ratio" << std::endl;
         std::cout << "ib:\t" << ib << "\n";
         std::cout << "temphisto->GetBinContent(ib):\t" << temphisto->GetBinContent(ib) << "\n";
         std::cout << "temphisto->GetBinError(ib):\t" << temphisto->GetBinError(ib) << "\n";
         tt = 100.;
       }
-      pe_spreads.push_back(tt);
-      // yL.push_back(me - tt);
-      // yM.push_back(me);
-      // yH.push_back(me + tt);
+      fRatioSpreads.push_back(tt);
     }
-    // peMax = *std::max_element(yH.begin(), yH.end());
-    // pe_l_InvSpl = TSpline3("pe_l_InvSpl", yL.data(), x.data(), yL.size());
-    // pe_m_InvSpl = TSpline3("pe_m_InvSpl", yM.data(), x.data(), yM.size());
-    // pe_h_InvSpl = TSpline3("pe_h_InvSpl", yH.data(), x.data(), yH.size());
     unsigned s = 0;
-    for(auto& peF : peFits){
-      std::string nold = "pe_fit_" + suffixes[s];
-      std::string nnew = "peFit_" + suffixes[s];
+    for(auto& ratioF : fRatioFits){
+      std::string nold = "ratio_fit_" + kSuffixes[s];
+      std::string nnew = "ratioFit_" + kSuffixes[s];
       TF1* tempF1 = (TF1*)infile->Get(nold.c_str());
       auto params = tempF1->GetParameters();
-      peF.f = std::make_unique<TF1>(nnew.c_str(), "pol3", 0., fDriftDistance);
-      peF.f->SetParameters(params);
-      peF.min = peF.f->GetMinimum(0., fDriftDistance, kEps);
-      peF.max = peF.f->GetMaximum(0., fDriftDistance, kEps);
+      ratioF.f = std::make_unique<TF1>(nnew.c_str(), fRatio_TF1_fit.c_str(),
+                                       0., fDriftDistance);
+      ratioF.f->SetParameters(params);
+      ratioF.min = ratioF.f->GetMinimum(0., fDriftDistance, kEps);
+      ratioF.max = ratioF.f->GetMaximum(0., fDriftDistance, kEps);
       s++;
     }
   }
 
+  TH2* tmp1_h2 = (TH2*)infile->Get("rr_h2");
+  fRRH2 = (TH2D*)tmp1_h2->Clone("fRRH2");
+  TH2* tmp2_h2 = (TH2*)infile->Get("ratio_h2");
+  fRatioH2 = (TH2D*)tmp2_h2->Clone("fRatioH2");
+
   infile->Close();
   delete infile;
   mf::LogInfo("FlashPredict") << "Finish loading metrics";
 }
 
 
-bool FlashPredict::computeChargeMetrics(const flashmatch::QCluster_t& qClusters)
+double FlashPredict::cheatMCT0(const std::vector<art::Ptr<recob::Hit>>& hits,
+                               const std::vector<art::Ptr<simb::MCParticle>>& mcParticles)
+{
+  auto clockData = art::ServiceHandle<detinfo::DetectorClocksService const>()->DataForJob();
+  const std::vector<std::pair<int, float>> matches =
```

Comment:
The truth match utils within LArSoft make this much simpler: https://github.com/LArSoft/larsim/blob/33c27d3378360a0b9fbb134263ac3e45221b58bc/larsim/Utils/TruthMatchUtils.h#L52-L62

Reply:
Very good advise, thanks I didn't know about it.